server/app/routes/user/index.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const router = require('express').Router();
+const User = mongoose.model('User');
+
+module.exports = router;
+
+// get all users
+router.get('/', (req, res, next) => {
+	User.find({})
+	.then(users => res.json(users))
+	.catch(next);
+});
+
+router.post('/', (req, res, next) => {
+	User.create(req.body)
+	.then(user => res.status(201).json(user))
+	.catch(next);
+});
+
+router.param('id', (req, res, next, id) => {
+	User.findbyId(id)
+	.then(user => {
+		req.requestUser = user;
+		next();
+	})
+	.catch(next);
+});
+
+router.put('/:id', (req, res, next) => {
+	req.requestUser.update(req.body)
+	.then(updatedUser => res.json(updatedUser))
```

Comment:
This looks like the user is returned (just one line in ES6 gets returned -- I think); however, using update circumvents pre and post save hooks.  If you have any pre or post save hooks, change this to set/save or alternatively, you can add pre/post "update" hooks.